src/Features/CSharp/Portable/UsePatternMatching/CSharpAsAndNullCheckDiagnosticAnalyzer.cs

```diff
@@ -95,6 +95,11 @@ private void SyntaxNodeAction(SyntaxNodeAnalysisContext syntaxContext)
             }
 
             var semanticModel = syntaxContext.SemanticModel;
+            if (semanticModel.GetSymbolInfo(comparison).GetAnySymbol().IsUserDefinedOperator())
```

Comment:
GetSymbolInfo returns a struct, so GetAnySymbol is safe.  IsUserDefinedOperator is one of our 'Is' extensions, and as such is null-safe.